### DISCUSSION TITLE
Computed columns

### DISCUSSION THREAD

**Discussion by mara-schulke** [SeaQL/sea-orm#1803] [Ideas]
## Motivation

Ive integrated sea orm in a quite complex project where the need arises for computed columns from different tables. 

## Proposed Solutions

A potential solution could be to have a computed column annotated on the model (which is **not** included in the derived active model) and just link a function which gets called on every selection of the model:

```rust
#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, DeriveEntityModel)]
#[sea_orm(table_name = "invoice_line_item")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i64,

    #[sea_orm(foreign_key)]
    pub invoice_id: i64,
     
    // ..  
  
    pub amount: i64,

    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}
```

```rust
#[derive(Clone, Debug, PartialEq, Eq, Serialize, Deserialize, DeriveEntityModel)]
#[sea_orm(table_name = "invoice")]
pub struct Model {
    #[sea_orm(primary_key)]
    pub id: i64,

    #[sea_orm(compute = "Entity::total")]
    pub total: i64,

    pub created_at: DateTime<Utc>,
    pub updated_at: DateTime<Utc>,
}

impl Entity {
    async fn total(pk: i64) -> Result<i64, sea_orm::error::DbErr> {
        // Run query which sums all related invoice line items
    }
}
```

## Additional Information

I think this would be pretty useful to a lot of realworld applications, what do you think about this? :slightly_smiling_face:

**Comment by ippsav** [comment]
This seems really helpful i would really like to have this as a feature as well !

**Comment by tyt2y3** [comment]
+1 for a PIP. I think the behaviour should be divided into on_insert and on_update (and on_change combines both) and the implementation should introduce a proper "prepare for insert/update" stage.

**Comment by mara-schulke** [comment]
@tyt2y3 thanks for seeing a usecase for this but as i understand queries of computed columns would not be ran at all oninsert / onupdate they are just there to combine seperate queries into the model on selection of a row.

**Comment by mara-schulke** [comment]
Any news? :) Id be glad to help but i dont where to start at all

**Comment by boringplay** [comment]
Any news? :)

**Comment by joleeee** [comment]
Is this possible to do now?
